src/coreclr/gc/gc.cpp

```diff
@@ -20110,40 +20117,69 @@ BOOL gc_heap::should_proceed_for_no_gc()
     BOOL no_gc_requested = FALSE;
     BOOL get_new_loh_segments = FALSE;
 
+    gc_heap* hp = nullptr;
     if (current_no_gc_region_info.soh_allocation_size)
     {
+#ifdef USE_REGIONS
 #ifdef MULTIPLE_HEAPS
         for (int i = 0; i < n_heaps; i++)
         {
-            gc_heap* hp = g_heaps[i];
-            if ((size_t)(heap_segment_reserved (hp->ephemeral_heap_segment) - hp->alloc_allocated) < hp->soh_allocation_no_gc)
+            hp = g_heaps[i];
+#endif //MULTIPLE_HEAPS
+            int result = hp->extend_soh_for_no_gc();
+            if (result == 1) // We ran out of reserved
```

Comment:
let's try to always avoid returning some random integer that then requires explanation like this. something like this -
```c
enum extend_soh_status
{
    short_on_reserve,
    short_on_commit
};

extend_soh_status gc_heap::extend_soh_for_no_gc();
```

Reply:
now I see you have a TODO in the header...

Reply:
however, thinking about this more, I don't see a point why you'd even need to make this distinction. the reason why we had this minimal_gc_p is so we could acquire a new segment for SOH. but since with regions we can freely acquire SOH regions without doing a GC we should just try to commit enough space on SOH (including getting new regions if needed) and if we can't, just do a full blocking GC.